### DISCUSSION TITLE
Lost data in view after click

### DISCUSSION THREAD

**Discussion by miquelangeld** [livewire/livewire#3250] [1. Help] [answered]
Hi. I'm testing livewire to migrate a vue website to livewire and i'm having some troubles.

I have a method which return a view `return view('checkin.booking',compact('clients'));` the clients data came from an external API, not a Laravel model.

In this view I call the livewire component like this `@livewire('get-booking',['clients' => $clients])`
And all is working fine, I get the render using some data in $clients.

But I created a button in the livewire component view with the click event like this `wire:click="clickPax()`

The method clickPax() just write a line in the log (i'm just testing). 

But after the  click I get an error "Trying to get property of nonobject" on the view. It seems as if after the click I have lost the data in the $clients variable. And I don't understand why.

My component:

`namespace App\Http\Livewire;

use Illuminate\Support\Facades\Log;
use Livewire\Component;

class GetBooking extends Component
{
    public $clients;

    public function render()
    {
        return view('livewire.get-booking');
    }

    public function clickPax(){
        Log::info('test');
    }
}`

**Comment by DarrylHardin** [comment]
add 
```
public function mount($clients)
{
   $this->clients = $clients;
}
```

**Reply by joshhanley** [reply]
From V2, LW automatically does this for you 🙂

**Reply by miquelangeld** [reply]
Thanks, finally the solution was (thanks to @joshhanley) convert the data structure (which was an array of objects) to a pure array

**Comment by joshhanley** [comment, marked as answer]
@miquelangeld Livewire only supports Eloquent Models/ Collections (as well as primitive types) as public properties as it needs to know how to serialise and deserialise them to send to the front end.

You said `$clients` comes from an api, what does that data look like? Are they objects? If so, try converting them to an array and then it should all work.

Hope this helps!

**Reply by miquelangeld** [reply]
Hi @joshhanley $clients is an array of objects. I converted all the data structure to array and now I don't get the error.
Thanks!

**Reply by miquelangeld** [reply]
I'm facing another problem @joshhanley  and maybe is related (if not, I can open another discussion) . f I pass data in the click event  `wire:click="clickPax({{$clients['booking']}})"`
the click does not work, I mean doesn't fire the post request. $clients['booking'] it's just a string.

**Reply by miquelangeld** [reply]
I answer my self. `clickPax('{{ $clients['Reserva'] }}', '{{ $booking['Linea'] }}')'"` I need to use quotes
